src/Service/CacheResetService.php

```diff
@@ -28,15 +28,23 @@ class CacheResetService
     protected $cacheResetIds = [];
 
     /**
+     * Should the complete cache be cleared after request
+     * @var bool
+     */
+    private $webhookCompleteReset = false;
+
+    /**
      * CacheResetService constructor.
+     *
      * @param CacheItemPoolInterface $cache
      * @param array $cacheResetIds
+     * @param bool $webhookCompleteReset
      */
-    public function __construct(CacheItemPoolInterface $cache, array $cacheResetIds)
+    public function __construct(CacheItemPoolInterface $cache, array $cacheResetIds, bool $webhookCompleteReset)
     {
-        $this
-            ->setCache($cache)
-            ->setCacheResetIds($cacheResetIds);
+        $this->setCache($cache)
+            ->setCacheResetIds($cacheResetIds)
+            ->setWebhookCompleteReset($webhookCompleteReset);
```

Comment:
To be honest i'am a bit confused here. Other Services in the bundle are using the setters and and in my opinion the pratice should be consistent across the same bundle.